webapp/src/serialindicator.tsx

```diff
@@ -19,9 +19,16 @@ export class SerialIndicator extends React.Component<SerialIndicatorProps, Seria
     constructor(props: any) {
         super(props)
         this.state = { active: false }
+    }
+
+    componentDidMount() {
         window.addEventListener("message", this.setActive.bind(this))
     }
 
+    componentWillUnmount() {
+        window.removeEventListener("message", this.setActive.bind(this))
+    }
+
     setActive(ev: MessageEvent) {
```

Comment:
This method _already_ verifies that `this.state.active` is false before setting it to true.  So it's not constantly updating the state as we thought.